src/createServer.js

```diff
@@ -1,3 +1,72 @@
-// Write code here
-// Also, you can create additional files in the src folder
-// and import (require) them here
+const http = require('http');
+const { convertToCase } = require('./convertToCase/convertToCase');
+
+const BASE = 'http://localhost:5701';
+
+const createServer = () => {
+  const server = http.createServer((req, res) => {
+    res.setHeader('Content-Type', 'application/json');
+
+    const reqUrl = new URL(BASE + req.url, BASE);
+    const originalText = reqUrl.pathname.slice(1);
+    const targetCase = reqUrl.searchParams.get('toCase');
+
+    const errors = [];
+
+    let conversionResult = {
+      originalCase: null,
+      convertedText: null,
+    };
+
+    if (!originalText) {
+      errors.push({
+        message: 'Text to convert is required. Correct request is: '
+        + '"/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".',
+      });
+    }
+
+    if (!targetCase) {
+      errors.push({
+        message: '"toCase" query param is required. Correct request is: '
+        + '"/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".',
+      });
+    } else {
+      try {
+        conversionResult = convertToCase(originalText, targetCase);
+      } catch {
+        errors.push({
+          message: 'This case is not supported. '
+          + 'Available cases: SNAKE, KEBAB, CAMEL, PASCAL, UPPER.',
+        });
+      }
+    }
+
+    const { originalCase, convertedText } = conversionResult;
+
+    if (errors.length) {
+      res.statusCode = 400;
+      res.statusMessage = 'Bad request';
+
+      res.write(JSON.stringify({
+        errors,
+      }));
+    } else {
+      res.statusCode = 200;
+
+      const resultObj = {
```

Comment:
Avoid using such abstract names, give a more detailed naming based on what will be stored in this object.